ARTEMIS-3573 Support PropertiesLoginModule custom password codecs (#3851)
apache/activemq-artemis: docs/user-manual/en/masking-passwords.md

With this configuration, both passwords in ra.xml and all of its MDBs will have
to be in masked form.

### PropertiesLoginModule
Artemis supports Properties login module to be configured in JAAS configuration file
(default name is `login.config`). By default, the passwords of the users are in plain text
or masked with the DefaultSensitiveStringCodec.

> Set the `org.apache.activemq.jaas.properties.password.codec` property to set a custom codec,
> i.e. to use the `org.apache.activemq.artemis.tests.integration.security.MD5SensitiveDataCodec` codec

Reviewer: ```suggestion
To use a custom codec class, set the `org.apache.activemq.jaas.properties.password.codec` property to the class name
e.g. to use the `com.example.MySensitiveDataCodecImpl` codec class:
```
Related to other comment, it definitely seems like we shouldnt be adding docs pointing to a class using MD5. Both since there seems little need using a real class (which is only part of the tests) at all, and since inevitably someone might actually go find it and just use what its doing.

Might also be worth linking to the section on custom codecs, since it wasnt covered yet and comes later.

```
PropertiesLoginWithPasswordCodec {
    org.apache.activemq.artemis.spi.core.security.jaas.PropertiesLoginModule required
        debug=true
        org.apache.activemq.jaas.properties.user="users.properties"
        org.apache.activemq.jaas.properties.role="roles.properties"

>         org.apache.activemq.jaas.properties.password.codec="org.apache.activemq.artemis.tests.integration.security.MD5SensitiveDataCodec";

Reviewer: ```suggestion
        org.apache.activemq.jaas.properties.password.codec="com.example.MySensitiveDataCodecImpl";
```

   public void init(Map<String, String> params) {
      // Initialization done here. It is called right after the codec has been created.
   }

   @Override
   public boolean compare(Object value, Object encodedValue) {

>       // Return true if the value matches the encodedValue.
>       return true;

Reviewer: ```suggestion
      // Return true if the value matches the encodedValue.
      return checkValueMatchesEncoding(value, encodedValue);
```

Just setting return true seems like a bad example, this at least spells it out.
